src/main/scala/slack/api/SlackApiClient.scala

```diff
@@ -1,66 +1,72 @@
+  private[api] implicit val fileInfoFmt          = Json.format[FileInfo]
   private[api] implicit val reactionsResponseFmt = Json.format[ReactionsResponse]
 
   val defaultSlackApiBaseUri = Uri("https://slack.com/api/")
 
   /* TEMPORARY WORKAROUND - UrlEncode '?' in query string parameters */
-  val charClassesClass = Class.forName("akka.http.impl.model.parser.CharacterClasses$")
-  val charClassesObject = charClassesClass.getField("MODULE$").get(charClassesClass)
+  val charClassesClass   = Class.forName("akka.http.impl.model.parser.CharacterClasses$")
+  val charClassesObject  = charClassesClass.getField("MODULE$").get(charClassesClass)
   //  strict-query-char-np
   val charPredicateField = charClassesObject.getClass.getDeclaredField("strict$minusquery$minuschar$minusnp")
   charPredicateField.setAccessible(true)
   val updatedCharPredicate = charPredicateField.get(charClassesObject).asInstanceOf[CharPredicate] -- '?'
   charPredicateField.set(charClassesObject, updatedCharPredicate)
   /* END TEMPORARY WORKAROUND */
 
-  def apply(token: String, slackApiBaseUri: Uri = defaultSlackApiBaseUri): SlackApiClient = {
-    new SlackApiClient(token, slackApiBaseUri)
+  def apply(token: String, slackApiBaseUri: Uri = defaultSlackApiBaseUri)(implicit system: ActorSystem): SlackApiClient = {
+    new SlackApiClient(token, slackApiBaseUri)(system)
   }
 
   def exchangeOauthForToken(
                              clientId: String,
                              clientSecret: String,
                              code: String,
                              redirectUri: Option[String] = None,
-                             slackApiBaseUri: Uri = defaultSlackApiBaseUri
+                             slackApiBaseUri: Uri = defaultSlackApiBaseUri,
+                             maybeSettings: Option[ConnectionPoolSettings] = None
                            )(implicit system: ActorSystem): Future[AccessToken] = {
     val params =
       Seq("client_id" -> clientId, "client_secret" -> clientSecret, "code" -> code, "redirect_uri" -> redirectUri)
-    val res = makeApiRequest(
-      addQueryParams(addSegment(HttpRequest(uri = slackApiBaseUri), "oauth.access"), cleanParams(params))
+    val res    = makeApiRequest(
+      addQueryParams(addSegment(HttpRequest(uri = slackApiBaseUri), "oauth.access"), cleanParams(params)),
+      maybeSettings
     )
     res.map(_.as[AccessToken])(system.dispatcher)
   }
 
-  private def makeApiRequest(request: HttpRequest)(implicit system: ActorSystem): Future[JsValue] = {
+  private def makeApiRequest(request: HttpRequest, maybeSettings: Option[ConnectionPoolSettings])(implicit system: ActorSystem): Future[JsValue] = {
     implicit val mat = ActorMaterializer()
-    implicit val ec = system.dispatcher
-    Http().singleRequest(request).flatMap {
+    implicit val ec  = system.dispatcher
+    Http().singleRequest(request, settings = maybeSettings.getOrElse(ConnectionPoolSettings(system))).flatMap {
```

Comment:
Not sure what is the Akka HTTP support for this, but even if there's none, can we add a wrapper around this, so that this `settings = ...` is not copied over and over again?